src/main/java/com/google/sps/servlets/LoginServlet.java

```diff
@@ -54,18 +57,33 @@ public void checkUserInDatastore(User user) {
     // ** NEED TO WRITE TEST FOR BEHAVIORAL TESTING
```

Comment:
This function 1) is not named properly and 2) does too many things.

1) The name of the function does not indicate to the programmer that it will store the user if they do not exist in the datastore.
2) A better way to write this logic would be to *not* add the user if they do not exist. Helper functions should generally do exactly one thing; this is currently doing two (checking and inserting).

So instead, you should make the function only check whether the user is in datastore, and then in your doGet, you could write

```
if (!isUserInDatastore(user)) {
  storeUserInDatastore(user);
}
```

Reply:
Ping?

Reply:
Should be fixed in #5 under "Restructure the json output in LoginServlet"